<?xml version="1.0" ?>
<launch>
    <!-- Realsense is assumed to be attached on b_bot -->
    <arg name="robot_name" default="a_bot"/>
    <arg name="camera_name" default="a_bot_camera" />
    <arg name="marker_size" default="0.16"
         doc="Size of the ArUco marker used, in meters" />

    <!-- Namespace for easy_handeye -->
    <arg name="namespace_prefix" value="o2as_easy_handeye_$(arg robot_name)" />

    <!-- Move group (TODO: check if it is actually required) -->
    <arg name="move_group" value="$(arg robot_name)" />

    <!-- Set AR marker ID -->
    <!-- We can use the same ID of different robots for eye-on-hand
    calibration -->
    <arg name="marker_id" value="32"/>

    <!-- Camera state publisher -->
    <!--<node pkg="robot_state_publisher" type="robot_state_publisher"
        name="d_bot_st_pub">
        <remap from="robot_description" to="d_bot_camera_description"/>
    </node>-->

    <!-- CameraInfo conversion for Phoxi in Gazebo:
         from camera_info to camera_info2,
         see convert_phoxi_camera_info_gazebo.py -->
    <node name="convert_phoxi_camera_info_gazebo" pkg="o2as_easy_handeye"
          type="convert_phoxi_camera_info_gazebo.py" />

    <!-- Start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <!-- Remap topic names -->
        <remap from="/camera_info" to="/$(arg camera_name)/rgb/camera_info"/>
        <remap from="/image" to="/$(arg camera_name)/rgb/image_raw" />

        <!-- Other parameters -->
        <param name="image_is_rectified" value="false"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="reference_frame"    value=""/>
        <param name="camera_frame"       value="camera_link"/>
        <param name="marker_frame"       value="camera_marker" />
    </node>

    <!-- Start Moveit (and Rviz) -->
    <include
        file="$(find o2as_moveit_config)/launch/o2as_moveit_planning_execution.launch">
        <arg name="sim" value="true" />
    </include>

    <!-- start easy_handeye calibrator -->
    <include file="$(find o2as_easy_handeye)/launch/calibrate_realsense.launch" >
        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
        <arg name="eye_on_hand" value="true" />

        <arg name="tracking_base_frame" value="camera_link" />
        <arg name="tracking_marker_frame" value="camera_marker" />
        <arg name="robot_base_frame" value="$(arg robot_name)_base_link" />
        <arg name="robot_effector_frame"
             value="$(arg robot_name)_cam_holder_link" />

        <arg name="freehand_robot_movement" value="true" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />

        <arg name="move_group" value="$(arg move_group)" />
    </include>

</launch>
